crates/datafusion_ext/src/runtime/table_provider.rs

```diff
@@ -21,13 +21,15 @@ use super::runtime_group::RuntimeGroupExec;
 pub struct RuntimeAwareTableProvider {
     pub preference: RuntimePreference,
     pub provider: Arc<dyn TableProvider>,
+    pub allow_writes: Option<bool>,
```

Comment:
this should definitely be an enum for "allowed_operations" (read, write_ddl, write_dml, write_all) or something like that. 

(there are potentially a few states for this, and the difference between false and not-specified-optional is confusing)

Reply:
> this should definitely be an enum for "allowed_operations" (read, write_ddl, write_dml, write_all) or something like that.

Agreed. Will update this and add tests.